File: CS320/HW1/3.hs
-- Teng Xu (xt@bu.edu)
-- 3.1
tet :: [Int] -> [Int]
tet(x) = sum (take 4 x) : x

--3.2
recur :: Int -> [Int]
recur 0 = [0]
recur n = recur(n-1) ++ [n]

--3.3
tet2help :: Int -> [Int] -> Int
tet2help n lst
    |n == 0 = 0
    |n == 1 = 1
    |n == 2 = 1
    |n == 3 = 2
    |n+1 == length lst = head lst
    |otherwise = tet2help n (tet lst)

tet2 :: Int -> Int
tet2 n = tet2help n [2,1,1,0]

File: CS320/HW1/3.py
# -*- coding: utf-8 -*-
"""
Created on Wed Sep 14 15:00:11 2016

@author: xt
"""
#3.1
def tet(lst):
    return [lst[0]+lst[1]+lst[2]+lst[3]]+lst

#3.2
def recur (n):
    if n == 0:
        return [0]
    else:
        return recur(n-1) + [n]

#3.3
def tet2(n):
    return tet2help(n, [2,1,1,0])

def tet2help(n, lst):
    if n == 0:
        return 0
    elif n == 1:
        return 1
    elif n == 2:
        return 1
    elif n == 3:
        return 2
    elif len(lst) == n+1:
        return lst[0]
    else:
        return tet2help(n, tet(lst))
